app/controllers/articles_controller.rb

```diff
@@ -84,6 +84,16 @@ def destroy
     redirect_to :articles
   end
 
+  def destroy_all
+    @article = current_user.articles
+    if @article&.destroy_all
+      flash[:success] = 'Your reset list has been reset.'
+    else
+      flash[:error] = 'There was an issue reseting your reading list.'
```

Comment:
Tiny typo here (`reseting` should be `resetting`), but might as well use `clearing` instead:

```suggestion
      flash[:error] = 'There was an issue clearing your reading list.'
```